File: 05/Java/TripleExpression/Variable.java
package TripleExpression;

public class Variable implements TripleExpression {
	private String name;
    public Variable(String nam) {
        name = nam;
    }

    public double evaluate(double x, double y, double z) {
        if (name.equals("x")) {
            return x;
        } else if (name.equals("y")) {
            return y;
        } else {
            return z;
        }
    }
}

File: 05/CSharp/ExpressionApp/Variable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExpressionApp
{
    class Variable : IExpression
    {
        private string name;
        public Variable(string name)
        {
            this.name = name;
        }
        public double Evaluate(double param)
        {
            return param;
        }
    }
}
